Khan/wonder-blocks
# WB-1636: Add `describedBy` prop to announce the popover dialog contents to screen readers.

## Summary:

The Popover dialog element is not assigning correctly the `aria-describedby`
value, thus causing Screen Readers not being able to read the popover contents
correctly.

The current implementation is trying to reference `aria-describedby` to the
anchor/trigger element (or button that triggers the popover to open/close). This
is incorrect, as the `aria-describedby` value should reference to a particular
content inside the dialog itself.

This PR adds a new prop to the Popover component, `describedBy`, which will
allow the user to pass a string referencing the element that contains the
dialog's contents. This will allow Screen Readers to read the popover contents
correctly.

Issue: https://khanacademy.atlassian.net/browse/WB-1636

## Test plan:

1. Navigate to /?path=/story/popover-popover--described-by.
2. Open the Popover.
3. Verify that the Screen Reader reads the title of the Popover dialog.
4. Change the `describedBy` prop to reference the `content` element by changing the
`describedBy` prop to `content` control in the Storybook.
5. Now verify that the SR reads the contents of the Popover dialog.
